File: Python/arrays/010_missing_number.py
# 010 — Missing Number

# Given array containing n distinct numbers in range [0,n], return missing one.

# Example

# Input: [3,0,1]
# Output: 2

input_nums = [3, 0, 1]
def missing_number(nums):
    n = len(nums)
    expected_sum = n * (n + 1) // 2
    actual_sum = sum(nums)
    
    return expected_sum - actual_sum
print(missing_number(input_nums))  # Output: 2

File: Go/arrays/010_missing_number.go
// 010 — Missing Number
// Given array containing n distinct numbers in range [0,n], return missing one.
// Example
// Input: [3,0,1]
// Output: 2

package arrays

func missingNumber(nums []int) int {
	n := len(nums)
	expectedSum := n * (n + 1) / 2
	actualSum := 0
	for _, num := range nums {
		actualSum += num
	}
	return expectedSum - actualSum
}
